OpenEdXMobile/src/main/java/org/edx/mobile/view/MainDiscoveryFragment.java

```diff
@@ -130,13 +130,19 @@ private void initFragments() {
     }
 
     private void onFragmentSelected(@IdRes int resId, final boolean isUserSelected) {
+        // First we need to hide all the fragments along with their shared search view,
+        // then show the required fragment and check either that fragment needs to show the
+        // search view or not through its `onHiddenChanged` method.
+        Fragment selectedFragment = null;
         for (int i = 0; i < fragmentsArray.size(); i++) {
```

Comment:
In my opinion it will be clean/safe approach if we write the block of this code in this way

```
        for (int i = 0; i < fragmentsArray.size(); i++) {
            hideFragment(fragmentsArray.valueAt(i));
            if (resId == fragmentsArray.keyAt(i)) {
                selectedFragment = fragmentsArray.valueAt(i);
            }
        }
```

Reply:
@farhan If we did with such code, the first fragment (Courses discovery) won't visible on the screen for Guest user search.